gpflow/experimental/check_shapes/__init__.py

```diff
@@ -139,11 +138,36 @@ def f(...):
         ...
 
 
+Checking intermediate results
+-----------------------------
+
+You can use the function :func:`check_shape` to check the shape of an intermediate result. This
+function will use the same namespace as the immediately surrounding :func:`check_shapes` decorator,
+and should only be called within functions that has such a decorator. For example::
+
+    @check_shapes(
+        "weights: [n_features, n_labels]",
+        "test_features: [n_rows, n_features]",
+        "test_labels: [n_rows, n_labels]",
+        "return: []",
+    )
+    def loss(
+        weights: np.ndarray, test_features: np.ndarray, test_labels: np.ndarray
+    ) -> np.ndarray:
+        prediction = check_shape(test_features @ weights, "[n_rows, n_labels]")
```

Comment:
Unrelated suggestion: make `config._enabled ` default to `False` if `__debug__` is `False` (so we emulate the behaviour of `assert`)?

Reply:
> Unrelated suggestion: make `config._enabled ` default to `False` if `__debug__` is `False` (so we emulate the behaviour of `assert`)?

Mmmm, dunno. Maybe. Definitely another PR, though.